src/jupyter_contrib_nbextensions/nbextensions/freeze/main.js

```diff
@@ -116,9 +118,14 @@ define([
                 continue;
             }
             var state = 'normal';
+            // Old metadata format
```

Comment:
Oh, I was thinking of something more like

```javascript
if (cell.metadata.run_control !== undefined && cell.metadata.run_control.read_only !== undefined) {
    if (cell.metadata.run_control.read_only) {
        cell.metadata.editable = false;
    }
    delete cell.metadata.run_control.read_only;
}
```

so that we adopt the new metadata just by loading the notebook - make sense?